CONTRIBUTION.md

```diff
@@ -63,6 +63,24 @@ We use [pytest](https://docs.pytest.org/en/latest/) for running tests. You can r
 poetry run pytest tests/
 ```
 
+There are two types of tests: those testing the internal functions and methods of the library and those testing the test execution with example queries. The latter require a running database instance to be available (either locally or remotely). Note, integration testing is currently only supported for Clickhouse. You can run internal testing only with:
+
+```bash
+poetry run pytest -m "not integration" tests/
+```
+
+Running integration tests locally requires [Docker](https://www.docker.com/) and [Docker Compose](https://docs.docker.com/compose/). First, start the local database services with:
+
+```bash
+docker compose up -d
+```
+
+Then you can run integration tests with:
+
+```bash
+poetry run pytest -m "integration" tests/
```

Comment:
**suggestion/discussion:** I am wondering whether we could simplify things and just use separate folders for the tests, e.g. 

* `tests/unit`
* `test/integration`

In the past I did some integration tests by running the tests in the examples folder. I like what you are proposing here, though. Only concern I have is the additional complexity with the markers since you would need to remember to add them. A folder structure seems easier to handle to me (but I am happy to discuss).

Reply:
Yeah I think I was referring to "adding the marker to each file" with my suggestion. The other thing can easily be solved with make files or other frameworks as you suggest.

But I also don't have a strong opinion against it. Would be nice if the markers also work in `__init__.py` files for the whole directory then (e.g. having an `test_integration` folder within clickhouse where you just put all of those if they get more than one file later).

For moving forward:
What if we try the markers but add a simple makefile to abstract things away?

Reply:
> Would be nice if the markers also work in __init__.py files for the whole directory then (e.g. having an test_integration folder within clickhouse where you just put all of those if they get more than one file later).

Good idea. I'm not sure whether this will work but I'll test it.

> What if we try the markers but add a simple makefile to abstract things away?

Sounds good. I'll add a `Makefile` to this PR.

Reply:
> I'm not sure whether this will work but I'll test it.

It does not ☹️